src/Swarm/TUI/Controller.hs

```diff
@@ -149,6 +151,16 @@ toggleModal s modal = do
   -- the world event handler for the KChar 'p'.
   resetLastFrameTime curTime = uiState . lastFrameTime .~ curTime
 
+handleModalEvent :: AppState -> V.Event -> EventM Name (Next AppState)
+handleModalEvent s ev = case ev of
+  V.EvKey V.KEnter [] ->
+    case s ^? uiState . uiModal . _Just . modalDialog . to dialogSelectedIndex of
```

Comment:
Using `dialogSelectedIndex` does not seem very type safe. Refering directly to a selected index leaks abstraction, since its semantics depends on the ordering of a list inside a subfunction of `generateModel` in `View.hs`. 
I propose replacing `dialogSelectedIndex` by `dialogSelection` which in this case is of type `Dialog Bool -> Maybe Bool` and gives us the state corresponding to the currently selected button. I would even propose going one step further and replacing `Bool` here by a custom `ButtonSelection` state because I imagine it to be an easy mistake to confuse whether `True` means cancel or quit.

Reply:
Ah, you're absolutely right.  I remember seeing `dialogSelection` when I was looking at the brick documentation but then I guess I forgot/didn't notice it when I actually wrote the code.  And your idea of replacing `Bool` with a custom `ButtonSelection` state is good, I will do that.